### DISCUSSION TITLE
asyncio vs cothreads

### DISCUSSION THREAD

**Discussion by Robatronic** [dls-controls/pythonSoftIOC#71] [General]
It seems to me that asyncio will work on both Windows and *nix.

Is there a compelling reason to use cothreads in Unix-land....?

**Comment by AlexanderWells-diamond** [comment]
My recent projects have exclusively used `asyncio` over `cothread`. 

From my understanding the main reason to stick to cothread is if you need to interface with code that already uses it, as trying to mix the two different systems can easily lead to a lot of mysterious, hard-to-debug, issues. 

You can also use `cothread` if you wish to avoid ever having to use `async/await` in your code.

**Reply by Robatronic** [reply]
asyncio should work under both Linux and Windows then.....?

**Reply by AlexanderWells-diamond** [reply]
Yes, asyncio is supported and tested on Windows and Linux as part of the CI, so everything should work fine.

**Comment by Araneidae** [comment]
Hi @Robatronic , hope you don't mind me moving your recent issues over here into the (newly opened) discussions section!

The distinction between `cothread` and `asyncio` is also largely historical.  `cothread` has been around a _lot_ longer than `asyncio` support in Python.

**Reply by Robatronic** [reply]
Not at all! Most of my inquiries are "discussion"-oriented. I'll use this more in future.  ;)

**Comment by thomascobb** [comment]
Also, cothread is an "implicit yield" module like gevent (which it also predates), while asyncio is an "explicit yield" module. Searching for "asyncio vs gevent" will give you more details on this topic:
https://piccolo-orm.com/blog/asyncio-vs-gevent/

**Comment by Robatronic** [comment]
This could be marked as answered.
